core/src/main/scala/org/broadinstitute/dsde/rawls/model/UserInfo.scala

```diff
@@ -4,22 +4,40 @@ import akka.http.scaladsl.model.headers.OAuth2BearerToken
 import com.google.api.client.auth.oauth2.Credential
 import com.google.auth.oauth2.GoogleCredentials
 
+import scala.util.Try
+
 /**
- * Created by dvoet on 7/21/15.
- */
-case class UserInfo(userEmail: RawlsUserEmail, accessToken: OAuth2BearerToken, accessTokenExpiresIn: Long, userSubjectId: RawlsUserSubjectId)
+  * Represents an authenticated user.
+  * @param userEmail the user's email address. Resolved to the owner if the request is from a pet.
+  * @param accessToken the user's access token. Either a B2C JWT or a Google opaque token.
+  * @param accessTokenExpiresIn number of seconds until the access token expires.
+  * @param userSubjectId the user id. Either a Google id (numeric) or a B2C id (uuid).
+  * @param googleAccessTokenThroughB2C if this is a Google login through B2C, contains the opaque
+  *                                    Google access token. Empty otherwise.
+  */
+case class UserInfo(userEmail: RawlsUserEmail,
```

Comment:
❓ Would it make sense to get a test around this logic and the other conditional B2C logic?

Reply:
Good call - this is probably unit-testable. Will add.